pkg/tests/apis/helper.go

```diff
@@ -205,10 +223,8 @@ func (c *K8sTestHelper) GetResourceClient(args ResourceClientArgs) *K8sResourceC
 		client, clientErr = dynamic.NewForConfig(args.User.NewRestConfig())
 	} else {
 		// Use service account token for authentication
-		cfg := &rest.Config{
-			Host:        fmt.Sprintf("http://%s", c.env.Server.HTTPServer.Listener.Addr()),
-			BearerToken: args.ServiceAccountToken,
-		}
+		cfg := newOptimizedRestConfigForTest(fmt.Sprintf("http://%s", c.env.Server.HTTPServer.Listener.Addr()))
```

Comment:
this change implies that service accounts are going to use the same config for tests... is that intended?

Reply:
maybe just `newOptimizedRestConfig` or `newOptimizedRestConfigForServiceAccounts`